### ISSUE TITLE
Cannot get S3 BucketPolicy created via CloudFormation

### ISSUE THREAD

**Issue by kestrelhawk** [localstack/localstack#1877]
I created a CloudFormation stack using the template below via Boto3 Python method `cf_client.create_stack(StackName="test")`; however, when I attempt to call `s3_client.get_bucket_policy(Bucket="my-bucket")` I receive a `NoSuchBucketPolicy ... The bucket policy does not exist` error. 

```yaml
---
AWSTemplateFormatVersion: "2010-09-09"
Description: A sample template

Resources:
  TestBucket:
    Type: AWS::S3::Bucket
    Properties:
      BucketName: my-bucket
      AccessControl: PublicRead
      BucketEncryption:
        ServerSideEncryptionConfiguration:
          - ServerSideEncryptionByDefault:
              SSEAlgorithm: AES256

  TestBucketPolicy:
    Type: AWS::S3::BucketPolicy
    Properties:
      Bucket:
        Ref: TestBucket
      PolicyDocument:
        Id: TestBucketPolicy
        Statement:
          - Action:
              - "s3:GetObject"
            Effect: "Allow"
            Resource:
              Fn::Join:
                - ""
                - - "arn:aws:s3:::"
                  - Ref: TestBucket
                  - "/*"
            Principal: "*"
            Condition:
              StringLike:
                aws:Referer:
                  - "http://www.example.com/*"
                  - "http://example.com/*"
```

Is this something that's supported? 

Thanks in advance.

**Comment by Mamoru1234** [comment]
As far as I know this kind of cloudformation functionality is not supported [code](https://github.com/localstack/localstack/blob/master/localstack/utils/cloudformation/template_deployer.py#L71). And if I understood correctly this kind of enhancement is non trivial to implement. Because boto client do not accept policy during creation :(

**Comment by whummer** [comment]
Thanks for reporting @kestrelhawk . This should be fixed in #2104 - can you please give it a try with the `latest` version. Please report here if the problem persists. Thanks
